trlx/trainer/accelerate_ppo_trainer.py

```diff
@@ -398,10 +408,21 @@ def make_experience(self, num_rollouts: int = 1024, iter_count: int = 0):  # noq
             else:
                 all_tokens = torch.cat((prompt_tensors.to(device), sample_outputs), dim=1)
                 attention_mask = all_tokens.not_equal(self.tokenizer.pad_token_id).long().to(device)
+                start_pos_idx = torch.argmax(
+                    torch.arange(attention_mask.shape[1], 0, -1, device=attention_mask.device) * attention_mask,
+                    1,
+                    keepdim=True,
+                )
+                position_ids = torch.max(
+                    torch.zeros_like(attention_mask),
+                    torch.arange(attention_mask.shape[1], device=attention_mask.device).repeat(
+                        (attention_mask.shape[0], 1)
+                    )
+                    - start_pos_idx,
+                )
```

Comment:
Do you think this computation can this be simplified down to:
```python
position_ids = attention_mask.long().cumsum(-1) - 1
position_ids.masked_fill_(attention_mask == 0, 1)
```
It seems to me that they are equivalent operationally (https://wandb.ai/sorry/trlx/reports/Different-position_ids-calculations--Vmlldzo0NzAzODg4)

Reply:
Yes that is a nicer way of doing it, thanks!